src/duk_debugger.c

```diff
@@ -1639,6 +1684,37 @@ DUK_LOCAL void duk__debug_process_message(duk_hthread *thr) {
 	return;
 }
 
+/* Halt execution and enter a debugger message loop until execution is resumed
+ * by the client. PC for the current activation may be temporarily decremented
+ * so that the "current" instruction will be highlighted in the client.
+ */
+
+DUK_INTERNAL void duk_debug_halt_execution(duk_hthread *thr, duk_bool_t use_prev_pc) {
+	duk_activation *act;
+	duk_instr_t *old_pc;
+
+	DUK_HEAP_SET_PAUSED(thr->heap);
+
+	/* Decrement PC if that was requested, this requires a PC sync */
+	act = duk_hthread_get_current_activation(thr);
+	duk_hthread_sync_currpc(thr);
+	old_pc = act->curr_pc;
+	if (use_prev_pc && act->curr_pc > 0) {
```

Comment:
The `act->curr_pc > 0` is now incorrect when `act->curr_pc` is a pointer and not a number. There's a helper to get the previous PC so perhaps this should be:

```
...
duk_hthread_sync_currpc(thr);
if (use_prev_pc) {
    old_pc = duk_hthread_act_get_prev_pc(thr, act);
} else {
    old_pc = duk_hthread_act_get_curr_pc(thr, act);
}
act->curr_pc = old_pc;
```

Not very clean I agree but should be correct.